packages/ui/src/components/va-virtual-scroller/useVirtualScrollerSizes.ts

```diff
@@ -0,0 +1,78 @@
+import { PropType, ExtractPropTypes, ref, Ref, shallowRef, computed, watch, onMounted } from 'vue'
+
+import { useResizeObserver } from '../../composables'
+
+export const useVirtualScrollerSizesProps = {
+  horizontal: { type: Boolean, default: false },
+  itemSize: {
+    type: [Number, String] as PropType<string | number>,
+    default: 0,
+    validator: (v: number | string) => {
+      if (typeof v === 'string') { return (!isNaN(+v) || v.endsWith('px') || v.endsWith('rem')) && parseInt(v) >= 0 }
```

Comment:
Would hope to see this logic being moved elsewhere (to utils maybe), but up to you.
Wondering if that's a concept we can reuse for other components.

Reply:
I can reuse it in split component.